### PR TITLE
process: update PR template to include release notes

### PR DESCRIPTION
**Pull request by 317brian**
### Description

* Updates the PR template to include a release notes section and a checkbox at the end
* Updates CONTRIBUTING.MD to include release notes tips and examples 

Followup work for this will include a PR to create a file in the repo to collect and copyedit the release notes from PRs on an ongoing basis to prepare for the next release. (I'll also need to do some catch up work for the existing PRs/commits that were added before this change gets implemented.)

We can add a stub file in this PR and direct people to add to it directly in lieu of them adding it to the PR description. But I think having people add it to the PR description would cause less friction based on how Druid release notes are currently oragnized.

<hr>

This PR has:
- [X] been self-reviewed.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -49,9 +58,11 @@ In each section, please describe design decisions made, including:
 <!-- Check the items by putting "x" in the brackets for the done things. Not all of these items apply to every PR. Remove the items which are not done or not relevant to the PR. None of the items from the checklist below are strictly necessary, but it would be very helpful if you at least self-review the PR. -->
 
 This PR has:
+
 - [ ] been self-reviewed.
    - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
 - [ ] added documentation for new or modified features or behaviors.
+- [ ] a release note entry added to the PR description.
```
Comment (techdocsmith) [inline .github/pull_request_template.md:65]:
```suggestion
- [ ] a release note entry in the PR description.
```
nit

**Review by techdocsmith — APPROVED** [review]
small nit. otherwise LGTM ✅

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -37,6 +37,15 @@ In each section, please describe design decisions made, including:
 
 <!-- Some of the aspects mentioned above may be omitted for simple and small changes. -->
 
+#### Release note
+<!-- Give your best effort to summarize your changes in a couple of sentences aimed toward Druid users. 
+
+If your change doesn't have end user impact, you can skip this section.
+
+-->
+For tips about how to write a good release note, see [Release notes](../CONTRIBUTING.md#release-notes).
```
Comment (clintropolis) [inline .github/pull_request_template.md:46]: the other links in the template are full links, I'm not certain if relative links work here or not, so to be safe we might want to make it match the others
```suggestion
For tips about how to write a good release note, see [Release notes](https://github.com/apache/druid/blob/master/CONTRIBUTING.md#release-notes).
```

**Review by clintropolis — COMMENTED** [review]
:+1:

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -139,6 +139,32 @@ You can find more developers' resources in [`dev/`](dev) directory.
   committer that merges your change will rebase and squash it into a single commit before
   committing it to master.
 
+## Release notes
+
+When you contribute a change, we want to make sure Druid users learn about the change. Give it your best shot! Druid committers will review them as part of the release process.
```
Comment (paul-rogers) [inline CONTRIBUTING.md:144]: Perhaps:

> Release notes are the way that Druid users will learn of your fix or improvement. What does a user need to know? Druid committers will review and edit your notes. The key is to identify the user impact.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -139,6 +139,32 @@ You can find more developers' resources in [`dev/`](dev) directory.
...
+
+**Summarize your change.** Describe the user impacting parts of your change, but try to be as concise as you can.
```
Comment (paul-rogers) [inline CONTRIBUTING.md:146]: Perhaps add:

> A change is user impacting if it:
>
> * Changes what the user sees in the UI.
> * Changes any action that the user takes (in the UI, in the API, in configuration, in install, etc.)
> * Changes the results of any query, ingestion or task.
> * Changes performance (preferably making things faster).
> * Adds or deprecates features.
> * Anything else that changes install, configuration or operation of Druid.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -139,6 +139,32 @@ You can find more developers' resources in [`dev/`](dev) directory.
...
+
+| Template                                        | Examples |
+|-------------------------------------------------|----------|
+| New: You can now…                           | New: You can now upload CSV files with a single header row for batch ingestion. |
+| Fixed: X no longer does Y when Z.               | Fixed: Multi-value string array expressions no longer give flattened results when used in  groupBy queries.         |
+| Changed: X now does Y. Previously, X did Z. | Changed: The first/last string aggregator now only compares based on values. Previously, the first/last string aggregator’s values were compared based on the `_time` column first and then on values.         |
+| Improved: Better / Increased / Updated etc. | Improved: You can now perform Kinesis ingestion even if there are empty shards. Previously, all shards had to have at least one record. |
+| Improved: Better / Increased / Updated etc. | Improved: Java 11 is fully supported and is no longer experimental. Java 17 support is improved.         |
+| Deprecated: Removed / Will remove X. | Deprecated: Support for ZooKeeper X.Y will be removed in the next release, so consider upgrading ZooKeeper.         |
```
Comment (paul-rogers) [inline CONTRIBUTING.md:165]: These are all great examples. The one item that might be missing is whether the user needs to do anything. For example, on that first one:

> New: You can now upload CSV files with a single header row for batch ingestion. Set the `infrerSchemaFromHeader` property of your Ingestion spec to `true` to enable this feature. See [doc link needed] for the details.

**Review by paul-rogers — CHANGES_REQUESTED** [review]
@317brian, thanks for writing this up! It will be quite helpful to both developers and (I hope) doc writers.

**Review by paul-rogers — APPROVED** [review]
@317brian, thanks for this improvement. LGTM.

**Pull request merged**